05-onchain.md

```diff
@@ -208,6 +208,9 @@ B's *commitment transaction*, hence the requirement to handle both.
 Each HTLC output can only be spent by us after it's timed out,
 or them if they have the payment preimage.
 
+There can be HTLCs which are not represented by an output: either
+because they were trimmed as dust, or in the case where B has two
+valid commitment transactions, and the HTLCs differ in each.
```

Comment:
> or in the case where B has two valid commitment transactions, and the HTLCs differ in each

Just to be sure: You mean when an htlc has already been fulfilled/failed, we just sent a new `commitment_signed`, and didn't yet received a `revoke_and_ack`. The counterparty has then two valid unrevoked tx, one with the htlc, one without.

Reply:
Exactly, and also it can happen the other way.  You could have added a new HTLC, sent commitment_signed and not received revoke_and_ack...